src/main/java/edu/byu/cs/util/RepoUrlValidator.java

```diff
@@ -1,24 +1,29 @@
 package edu.byu.cs.util;
 
 import edu.byu.cs.autograder.GradingException;
+import edu.byu.cs.autograder.git.GitHelper;
 import org.eclipse.jgit.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import java.io.File;
+import java.util.UUID;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class RepoUrlValidator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RepoUrlValidator.class);
 
-    public static boolean isValid(@Nullable String repoUrl) {
+    public static boolean isValidRepoUrl(String url) {
+        File cloningDir = new File("./tmp" + UUID.randomUUID());
+        boolean valid = true;
         try {
-            clean(repoUrl);
-            return true;
-        } catch (InvalidRepoUrlException e) {
-            return false;
+            GitHelper.fetchRepo(cloningDir, url);
```

Comment:
It already is using the now extracted method of GitHelper.
And, for the 4 cases that you provided, I'm confused on which ones are failing that shouldn't or passing when they should fail? Your examples follow what should already happen, no? Because the current chain of code calls clean and validifies it (updateRepoUrl calls clean and setRepoUrl, which calls isValidRepoUrl (although I could rename it to isValid))